<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow2</class>
 <widget class="QMainWindow" name="MainWindow2">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1300</width>
    <height>730</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="mainOpenglWindow2" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>1010</width>
      <height>700</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>20</y>
      <width>140</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Environment Name :</string>
    </property>
   </widget>
   <widget class="QPushButton" name="browseEnvButton">
    <property name="geometry">
     <rect>
      <x>1181</x>
      <y>10</y>
      <width>98</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Browse</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>111</y>
      <width>140</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Name :</string>
    </property>
   </widget>
   <widget class="QPushButton" name="browseRobotButton">
    <property name="geometry">
     <rect>
      <x>1181</x>
      <y>101</y>
      <width>98</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Browse</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>90</y>
      <width>66</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Robot:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="initRobotButton">
    <property name="geometry">
     <rect>
      <x>1101</x>
      <y>190</y>
      <width>180</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Initialise Robot Position</string>
    </property>
   </widget>
   <widget class="QPushButton" name="locationResetButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1114</x>
      <y>354</y>
      <width>60</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>230</y>
      <width>21</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>X :</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="locationXspinBox">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>250</y>
      <width>64</width>
      <height>28</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>2700</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>20</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="locationYspinBox">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1101</x>
      <y>250</y>
      <width>64</width>
      <height>28</height>
     </rect>
    </property>
    <property name="minimum">
     <number>10</number>
    </property>
    <property name="maximum">
     <number>2700</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>1101</x>
      <y>230</y>
      <width>21</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Y :</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>200</y>
      <width>71</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Location :</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>1171</x>
      <y>230</y>
      <width>21</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Z :</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="locationZspinBox">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1171</x>
      <y>250</y>
      <width>64</width>
      <height>28</height>
     </rect>
    </property>
    <property name="minimum">
     <number>20</number>
    </property>
    <property name="maximum">
     <number>2700</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
   </widget>
   <widget class="QPushButton" name="locationUndoButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1034</x>
      <y>354</y>
      <width>60</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Undo</string>
    </property>
   </widget>
   <widget class="QPushButton" name="locationSaveButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1194</x>
      <y>354</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Save</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="locationAnglespinBox">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>310</y>
      <width>64</width>
      <height>28</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-360</number>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>290</y>
      <width>50</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>Angle :</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="envNamelineEdit">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>50</y>
      <width>250</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="robotNamelineEdit">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>144</y>
      <width>250</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="startButton">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>400</y>
      <width>240</width>
      <height>80</height>
     </rect>
    </property>
    <property name="text">
     <string>Start</string>
    </property>
   </widget>
   <widget class="QLabel" name="statuslabel">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>560</y>
      <width>240</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>. . .</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="leftSpinBox">
    <property name="geometry">
     <rect>
      <x>1429</x>
      <y>620</y>
      <width>62</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-300.000000000000000</double>
    </property>
    <property name="maximum">
     <double>300.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
    <property name="value">
     <double>10.000000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="FpushButton">
    <property name="geometry">
     <rect>
      <x>1350</x>
      <y>590</y>
      <width>30</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>F</string>
    </property>
   </widget>
   <widget class="QPushButton" name="LpushButton">
    <property name="geometry">
     <rect>
      <x>1320</x>
      <y>620</y>
      <width>30</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>L</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rightSpinBox">
    <property name="geometry">
     <rect>
      <x>1500</x>
      <y>620</y>
      <width>62</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-300.000000000000000</double>
    </property>
    <property name="maximum">
     <double>300.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
    <property name="value">
     <double>10.000000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="RpushButton">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>620</y>
      <width>30</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>R</string>
    </property>
   </widget>
   <widget class="QPushButton" name="BpushButton">
    <property name="geometry">
     <rect>
      <x>1350</x>
      <y>650</y>
      <width>30</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>B</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>1430</x>
      <y>650</y>
      <width>131</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>580</y>
      <width>110</width>
      <height>27</height>
     </rect>
    </property>
    <property name="maximum">
     <double>500.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>5.000000000000000</double>
    </property>
    <property name="value">
     <double>10.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>590</y>
      <width>90</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Velocity</string>
    </property>
   </widget>
   <widget class="QPushButton" name="resetButton">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>533</y>
      <width>240</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stopButton">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>493</y>
      <width>240</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1300</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuAuthor">
     <property name="title">
      <string>Author</string>
     </property>
     <addaction name="actionAtul_Rokade_IITB"/>
    </widget>
    <addaction name="actionEnvironment"/>
    <addaction name="actionRobot_2"/>
    <addaction name="separator"/>
    <addaction name="menuAuthor"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionEnviourment">
   <property name="text">
    <string>Enviourment</string>
   </property>
  </action>
  <action name="actionRobot">
   <property name="text">
    <string>Robot</string>
   </property>
  </action>
  <action name="actionEnvironment">
   <property name="text">
    <string>Enviourment</string>
   </property>
  </action>
  <action name="actionRobot_2">
   <property name="text">
    <string>Robot</string>
   </property>
  </action>
  <action name="actionAtul_Rokade_IITB">
   <property name="text">
    <string>Atul Rokade (IITB)</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>mainOpenglWindow2</class>
   <extends>QWidget</extends>
   <header>mainopenglwindow2.h</header>
   <container>1</container>
   <slots>
    <slot>getEnvName(string)</slot>
    <slot>getRobotName(string)</slot>
    <slot>getPortNumber(int)</slot>
    <slot>start()</slot>
    <slot>updateRobotLocation(robotLocation)</slot>
    <slot>forward()</slot>
    <slot>backward()</slot>
    <slot>left()</slot>
    <slot>right()</slot>
    <slot>setVelocityLeft(double)</slot>
    <slot>setVelocityRight(double)</slot>
    <slot>stop()</slot>
    <slot>setVelocity(double)</slot>
    <slot>reset()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>menubar</sender>
   <signal>triggered(QAction*)</signal>
   <receiver>MainWindow2</receiver>
   <slot>menuClicked(QAction*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>45</x>
     <y>12</y>
    </hint>
    <hint type="destinationlabel">
     <x>151</x>
     <y>34</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow2</sender>
   <signal>setEnvName(string)</signal>
   <receiver>widget</receiver>
   <slot>getEnvName(string)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>428</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>292</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>browseEnvButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow2</receiver>
   <slot>browseEnvButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1225</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>1148</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>browseRobotButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow2</receiver>
   <slot>browseRobotButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1242</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>1206</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>initRobotButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow2</receiver>
   <slot>initRobotButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1212</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>1175</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationUndoButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow2</receiver>
   <slot>undoButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1047</x>
     <y>389</y>
    </hint>
    <hint type="destinationlabel">
     <x>1054</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationResetButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow2</receiver>
   <slot>resetButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1129</x>
     <y>387</y>
    </hint>
    <hint type="destinationlabel">
     <x>1134</x>
     <y>362</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationSaveButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow2</receiver>
   <slot>locationSaveButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1237</x>
     <y>388</y>
    </hint>
    <hint type="destinationlabel">
     <x>1258</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationXspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow2</receiver>
   <slot>locationXspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1054</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel">
     <x>1067</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationYspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow2</receiver>
   <slot>locationYspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1126</x>
     <y>287</y>
    </hint>
    <hint type="destinationlabel">
     <x>1133</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationZspinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow2</receiver>
   <slot>locationZspinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1195</x>
     <y>281</y>
    </hint>
    <hint type="destinationlabel">
     <x>1248</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationAnglespinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow2</receiver>
   <slot>locationAnglespinBoxValueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1086</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>1116</x>
     <y>346</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow2</sender>
   <signal>setEnvName(string)</signal>
   <receiver>widget</receiver>
   <slot>getEnvName(string)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1283</x>
     <y>110</y>
    </hint>
    <hint type="destinationlabel">
     <x>877</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow2</sender>
   <signal>setRobotName(string)</signal>
   <receiver>widget</receiver>
   <slot>getRobotName(string)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1287</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>997</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1120</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>952</x>
     <y>531</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow2</sender>
   <signal>setRobotLocation(robotLocation)</signal>
   <receiver>widget</receiver>
   <slot>updateRobotLocation(robotLocation)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1076</x>
     <y>463</y>
    </hint>
    <hint type="destinationlabel">
     <x>998</x>
     <y>467</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LpushButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>left()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1320</x>
     <y>671</y>
    </hint>
    <hint type="destinationlabel">
     <x>967</x>
     <y>661</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>FpushButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>forward()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1350</x>
     <y>641</y>
    </hint>
    <hint type="destinationlabel">
     <x>965</x>
     <y>632</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RpushButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>right()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1380</x>
     <y>671</y>
    </hint>
    <hint type="destinationlabel">
     <x>930</x>
     <y>677</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BpushButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>backward()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1350</x>
     <y>701</y>
    </hint>
    <hint type="destinationlabel">
     <x>938</x>
     <y>695</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>leftSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setVelocityLeft(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1429</x>
     <y>671</y>
    </hint>
    <hint type="destinationlabel">
     <x>922</x>
     <y>609</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rightSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setVelocityRight(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1500</x>
     <y>671</y>
    </hint>
    <hint type="destinationlabel">
     <x>969</x>
     <y>705</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1430</x>
     <y>701</y>
    </hint>
    <hint type="destinationlabel">
     <x>967</x>
     <y>592</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>widget</receiver>
   <slot>setVelocity(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1181</x>
     <y>631</y>
    </hint>
    <hint type="destinationlabel">
     <x>958</x>
     <y>573</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1097</x>
     <y>563</y>
    </hint>
    <hint type="destinationlabel">
     <x>999</x>
     <y>554</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1117</x>
     <y>526</y>
    </hint>
    <hint type="destinationlabel">
     <x>1006</x>
     <y>523</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>setEnvName(string)</signal>
  <signal>setRobotName(string)</signal>
  <signal>setRobotLocation(robotLocation)</signal>
  <slot>menuClicked(QAction*)</slot>
  <slot>browseEnvButtonClicked()</slot>
  <slot>browseRobotButtonClicked()</slot>
  <slot>initRobotButtonClicked()</slot>
  <slot>undoButtonClicked()</slot>
  <slot>resetButtonClicked()</slot>
  <slot>locationSaveButtonClicked()</slot>
  <slot>locationXspinBoxValueChanged(int)</slot>
  <slot>locationYspinBoxValueChanged(int)</slot>
  <slot>locationZspinBoxValueChanged(int)</slot>
  <slot>locationAnglespinBoxValueChanged(int)</slot>
  <slot>portNumberspinBoxValueChanged(int)</slot>
  <slot>startButtonClicked()</slot>
 </slots>
</ui>
